test/e2e/e2e_test.go

```diff
@@ -131,6 +133,159 @@ var _ = ginkgo.Describe("JobSet", func() {
 		})
 	})
 
+	ginkgo.When("job is unsuspended and suspend", func() {
+		ginkgo.It("should not create Jobs while suspended, and delete Jobs on suspend", func() {
+			ctx := context.Background()
+			js := shortSleepTestJobSet(ns).Obj()
+			jsKey := types.NamespacedName{Name: js.Name, Namespace: js.Namespace}
+
+			ginkgo.By("Create a suspended JobSet", func() {
+				js.Spec.Suspend = ptr.To(true)
+				js.Spec.TTLSecondsAfterFinished = ptr.To[int32](5)
```

Comment:
It's not necessary, I just added it so that the JobSets get removed eventually from the API server, but it would all work equally well without it. I can remove. WDYT?